core/src/main/java/org/infinispan/marshall/exts/CollectionExternalizer.java

```diff
@@ -78,50 +69,32 @@ public void writeObject(ObjectOutput output, Collection collection) throws IOExc
          case SYNCHRONIZED_SET:
          case SYNCHRONIZED_LIST:
          case ARRAY_DEQUE:
-         case READ_ONLY_SEGMENT_AWARE_COLLECTION:
-            MarshallUtil.marshallCollection(collection, output);
-            break;
          case SINGLETON_LIST:
             output.writeObject(((List) collection).get(0));
             break;
          case SINGLETON_SET:
             output.writeObject(collection.iterator().next());
             break;
-         case TREE_SET:
```

Comment:
I don't think TREE_SET will work as it has a guaranteed ordering, which copying into a HashSet will not guarantee.

Reply:
Ack. I wanted to see where we actually depended on TreeSet, but it's required by Multimap.